plugins/inventory/nb_inventory.py

```diff
@@ -638,6 +684,39 @@ def extract_custom_fields(self, host):
         except Exception:
             return
 
+    def extract_regions(self, host):
+        # A host may have a site. A site may have a region. A region may have a parent region.
+        # Produce a list of regions:
+        # - it will be empty if the device has no site, or the site has no region set
+        # - it will have 1 element if the site's region has no parent
+        # - it will have multiple elements if the site's region has a parent region
+
+        site = host.get("site", None)
+        if not isinstance(site, dict):
+            # Device has no site
+            return []
+
+        site_id = site.get("id", None)
+        if site_id is None:
+            # Device has no site
+            return []
+
+        regions = []
+        region_id = self.sites_region_lookup[site_id]
+
+        # Keep looping until the region has no parent
+        while region_id is not None:
+            region_slug = self.regions_lookup[region_id]
+            if region_slug in regions:
```

Comment:
I tested this and appears it doesn't get stuck in a loop, but maybe my test data isn't correct for L710-L712

Reply:
No this will never get stuck in a loop – wasn't sure how defensively I should be programming against the Netbox API. It would only be in theory if Netbox had a corrupt database that somehow had a loop in it.

Reply:
Ok no problem. I don't feel strongly on either having it or not so we can leave it.

Reply:
I've left this in but updated the comment to be more clear.